ts/parts/LogarithmicAxis.ts

```diff
@@ -19,180 +26,267 @@ import H from './Globals.js';
+                tickIntervalOption = minor ?
+                    axis.getMinorTickInterval() :
+                    options.tickInterval,
+                filteredTickIntervalOption = tickIntervalOption === 'auto' ?
+                    null :
+                    tickIntervalOption,
+                tickPixelIntervalOption =
+                    (options.tickPixelInterval as any) / (minor ? 5 : 1),
+                totalPixelLength = minor ?
+                    axisLength / axis.tickPositions.length :
+                    axisLength;
+
+            interval = pick(
+                filteredTickIntervalOption,
+                log.minorAutoInterval,
+                (realMax - realMin) *
+                    tickPixelIntervalOption / (totalPixelLength || 1)
+            );
+
+            interval = normalizeTickInterval(
+                interval,
+                void 0,
+                getMagnitude(interval)
+            );
+
+            positions = axis.getLinearTickPositions(
+                interval,
+                realMin,
+                realMax
+            ).map(axis.log2lin);
 
+            if (!minor) {
+                log.minorAutoInterval = interval / 5;
+            }
+        }
+
+        // Set the axis-level tickInterval variable
         if (!minor) {
-            axis._minorAutoInterval = interval / 5;
+            axis.tickInterval = interval;
         }
+        return positions;
     }
 
-    // Set the axis-level tickInterval variable
-    if (!minor) {
-        axis.tickInterval = interval;
+    public lin2log(num: number): number {
+        const axis = this.axis;
+        const lin2log = axis.options.linearToLogConverter;
+
+        if (typeof lin2log === 'function') {
+            return lin2log.apply(axis, arguments);
+        }
+
+        return Math.pow(10, num);
     }
-    return positions;
-};
 
-/**
- * @private
- * @function Highcharts.Axis#log2lin
- *
- * @param {number} num
- *
- * @return {number}
- */
-Axis.prototype.log2lin = function (this: Highcharts.Axis, num: number): number {
-    return Math.log(num) / Math.LN10;
-};
+    public log2lin(num: number): number {
+        return Math.log(num) / Math.LN10;
+    }
 
-/**
- * @private
- * @function Highcharts.Axis#lin2log
- *
- * @param {number} num
- *
- * @return {number}
- */
-Axis.prototype.lin2log = function (this: Highcharts.Axis, num: number): number {
-    return Math.pow(10, num);
-};
+}
+
+class LogarithmicAxis {
+
+    /**
+     * Provides logarithmic support for axes.
+     *
+     * @private
+     */
+    public static compose(AxisClass: typeof Axis): void {
+
+        const axisProto = AxisClass.prototype as LogarithmicAxis;
+
+        /**
+         * @deprecated
+         * @private
+         * @function Highcharts.Axis#lin2log
+         *
+         * @param {number} num
+         *
+         * @return {number}
+         */
+        axisProto.lin2log = function (this: Highcharts.Axis, num: number): number {
```

Comment:
The to converter functions are duplicated, defined both on the LogarithmicAxisAdditions and on the Axis itself. We need to avoid that. Either one of them need to call the other (probably best if the `axis` calls `axis.logarithmic.lin2log`), or one of them need to be borrowed from the other. Like `axisProto.lin2log = LogarithmicAxisAdditions.prototype.lin2log`.

Reply:
.. Or define local functions that are set on both classes.

Reply:
I do this because the `this` scope is different, so you would need wrapper functions anyway. You can not simply assign them.